src/lv_widgets/lv_chart.c

```diff
@@ -755,30 +756,25 @@ lv_point_t lv_chart_get_cursor_point(lv_obj_t * chart, lv_chart_cursor_t * curso
 }
 
 /**
- * Get the nearest index we have in the left side of a point on series area.
+ * Get the nearest index to an X coordinate
  * @param chart pointer to a chart object
- * @param coord the coordination of the point.
- * @return the index in found.
+ * @param coord the coordination of the point relative to the series area.
+ * @return the found index
  */
-uint16_t lv_chart_get_nearest_index_from_coord(lv_obj_t * chart, lv_point_t coord)
+uint16_t lv_chart_get_nearest_index_from_coord(lv_obj_t * chart, lv_coord_t x)
 {
     lv_area_t series_area;
     lv_chart_get_series_area(chart, &series_area);
 
     lv_coord_t w = lv_area_get_width(&series_area);
-
-    uint16_t id = 0;
     lv_chart_ext_t * ext = lv_obj_get_ext_attr(chart);
 
-    if(coord.x < 0) {
-	    id = 0;
-    } else if(coord.x > w) {
-	    id = ext->point_cnt - 1;
-    } else {
-	    id = coord.x * (ext->point_cnt - 1) / w;
-    }
+    if(x < 0) return 0;
+    if(x > w) return ext->point_cnt - 1;
```

Comment:
Just asking:
Why not writing like this:
```
if(x < 0) return 0;
else if(x > w) return ext->point_cnt - 1;
```

Reply:
It's kind of subjective but I "believe" in early returns instead of a single return point.
Therefore I return when I can. :slightly_smiling_face:

Reply:
OK, so good,
> I "believe" in early returns instead of a single return point.
Hmm, It reminds me Misra C.
By the way, Have you ever thought about passing Misra C?

Reply:
I was thinking about it, but now I think that the restrictions are more than the advantages. :slightly_smiling_face: